src/usr/share/netcheck/netcheck.py

```diff
@@ -127,296 +133,295 @@ def _dns_works(self, network):
-                self.connected_wifi_index = index
+            if wireless_activation_successful:
+                self.logger.debug(
+                    '_try_wireless_connections: wireless connection %s is activated with'
+                    'successful DNS check.', connection_id)
+                self.activated_wireless_index = index
                 success = True
 
             else:
-                self.logger.debug('_try_wifi_networks: Wifi network %s failed to connect or '
-                                  'failed DNS check.' % network_name)
-                success = self._try_wifi_networks(index + 1)
+                self.logger.debug(
+                    '_try_wireless_connections: wireless connection %s failed to activate '
+                    'or failed DNS check.', connection_id)
+                success = self._try_wireless_connections(index + 1)
 
         return success
 
-    # TODO #11: Consider downloading a small file upon successful connection so we are sure
-    #   FreedomPop considers this network used.
-    # TODO #12: The random interval should probably be applied after the last DNS check on the
-    #   backup network. (We might have used the backup network since the last check or
-    #   might even be currently connected to it.)
-    def _use_backup_network(self):
-        """Use the highest-priority wireless network randomly between zero and a
-        user-specified number of days. Recalculate that interval every time the network is
-        checked and then call _try_wifi_networks. FreedomPop requires monthly usage and is
-        assumed to be the highest-priority network.
+    # TODO: Consider downloading a small file upon successful connection so we are sure
+    #   FreedomPop considers this connection used. (issue 11)
+    # TODO: The random interval should probably be applied after the last DNS check on the
+    #   backup connection. (We might have used the backup connection since the last check or
+    #   the backup connection might even be currently activated.) (issue 12)
+    def _use_backup_connection(self):
+        """Use the highest-priority wireless connection randomly between zero and a
+        user-specified number of days.  Recalculate that interval every time the connection
+        is checked and then call _try_wireless_connections.  FreedomPop requires monthly
+        usage and is assumed to be the highest-priority connection.
         """
+        self.logger.trace(
+            '_use_backup_connection: Determining if the main wireless backup connection '
+            'should be used.')
 
-        self.logger.trace('_use_backup_network: Determining if we should use the main WiFi '
-                          'backup network.')
-
-        if datetime.datetime.now() >= self.backup_network_check_time:
-
-            self.logger.info('Trying to use backup wifi network.')
+        if datetime.datetime.now() < self.backup_connection_check_time:
```

Comment:
We might want to put a try/except block here so we don't get stuck on a single connection.